app/src/main/java/com/amaze/filemanager/asynchronous/services/ProgressiveServiceAbstract.java

```diff
@@ -0,0 +1,277 @@
+package com.amaze.filemanager.asynchronous.services;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+import android.text.format.Formatter;
+import android.util.Log;
+
+import com.amaze.filemanager.R;
+import com.amaze.filemanager.activities.MainActivity;
+import com.amaze.filemanager.filesystem.HybridFile;
+import com.amaze.filemanager.fragments.ProcessViewerFragment;
+import com.amaze.filemanager.ui.notifications.NotificationConstants;
+import com.amaze.filemanager.utils.DatapointParcelable;
+import com.amaze.filemanager.utils.ProgressHandler;
+import com.amaze.filemanager.utils.ServiceWatcherUtil;
+
+import java.util.ArrayList;
+
+/**
+ * @author Emmanuel Messulam <emmanuelbendavid@gmail.com>
+ *         on 28/11/2017, at 19:32.
+ */
+
+public abstract class ProgressiveServiceAbstract extends Service implements ServiceWatcherUtil.ServiceWatcherInteractionInterface {
```

Comment:
I still don't understand the use of ` ServiceWatcherUtil.ServiceWatcherInteractionInterface`, you can make the functions abstract and do away with the interface.

Reply:
````ServiceWatcherUtil```` uses it to interact with the pause/resume of service.

Reply:
It is passing the `ProgressiveService` as a `ServiceWatcherUtil.ServiceWatcherInteractionInterface`, it can use it directly (if you move the methods).

Reply:
No it can't. There's no need to pass around whole service when you need only some part of it.

Reply:
Ok, pass it as an instance of the interface, so as to make more difficult unwanted access to other functions.